### PR TITLE
Introduce method shorthands for Request constructor

### PR DESCRIPTION
**Pull request by zeroed**
Expose easy shorthands for Request constructors to match the approach
used in Surf.

## Checklist

- [x] introduce method shorthands for Request constructor;
- [x] add tests;
- [x] update documentation and examples;

Refs: #156

### CODE REVIEW

File: src/request.rs [hunk 1]
```diff
@@ -648,10 +665,110 @@ impl Request {
...
+        U: TryInto<Url>,
+        U::Error: std::fmt::Debug,
+    {
+        Request::new(Method::Connect, url)
+    }
+
+    /// Create a OPTIONS request.
+    ///
+    /// # Examples
+    pub fn optiond<U>(url: U) -> Self
```
Comment (Fishrock123) [inline src/request.rs:738]:
```suggestion
    pub fn options<U>(url: U) -> Self
```

Comment (zeroed) [reply src/request.rs:738]: :+1: 
Thank you

**Comment by Fishrock123** [comment]
Fwiw seems fine to me.

**Comment by zeroed** [comment]
Thank you.
I will ensure `cargo fmt`, tests (and no typos) and mark the PR as "Ready for review".

**Pull request merged**